core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorResponseFactory.java

```diff
@@ -79,14 +66,14 @@ else if (isAssignable(RuleViolationException.class, cursor))
             }
             else if (isAssignable(ClientErrorException.class, cursor))
             {
-                result = errorMapper.handleClientError((ClientErrorException) cursor);
+                result = clientErrorMapper.handleClientError((ClientErrorException) cursor);
```

Comment:
I wonder why we (even before this PR) `break` if `genericErrorMapper` found something, but don't do the same for the other `isAssignable()` checks. Why do we continue iterating the `cause` chain, overwriting a response `result` we already determined? I can only imagine that this bug just doesn't occur in practice because `ConstraintViolationException`, `RuleViolationException` and `ClientErrorException` are unlikely to even have cause exceptions. Still, we should be consistent here.

Please also add a unit test (for at least one case) that verifies we stop iterating the `cause` chain as soon as we have a non-null behavior or `handle*()` result.